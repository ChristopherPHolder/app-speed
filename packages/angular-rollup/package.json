{
  "name": "@app-speed/angular-rollup",
  "version": "0.0.1",
  "dependencies": {
    "@nx/devkit": "18.2.1",
    "esbuild": "^0.19.2",
    "rollup": "^4.18.0"
  },
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts",
  "private": true,
  "executors": "./executors.json"
}
