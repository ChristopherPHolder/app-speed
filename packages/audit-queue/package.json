{
  "name": "@app-speed/audit-queue",
  "version": "0.0.1"
}
