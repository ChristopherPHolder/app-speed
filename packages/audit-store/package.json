{
  "name": "@app-speed/audit-store",
  "version": "0.0.1"
}
