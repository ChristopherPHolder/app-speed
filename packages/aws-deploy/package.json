{
  "name": "@app-speed/aws-deploy",
  "version": "0.0.1",
  "dependencies": {
    "@nx/devkit": "19.8.2",
    "@aws-sdk/client-lambda": "^3.609.0",
    "dotenv": "16.4.5",
    "archiver": "^7.0.1"
  },
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts",
  "executors": "./executors.json"
}
