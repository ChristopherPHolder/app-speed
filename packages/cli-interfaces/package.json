{
  "name": "@app-speed/cli-interfaces",
  "version": "0.0.1"
}
