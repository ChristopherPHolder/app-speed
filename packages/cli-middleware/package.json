{
  "name": "@app-speed/cli-middleware",
  "version": "0.0.1"
}
