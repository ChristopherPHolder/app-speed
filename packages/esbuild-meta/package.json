{
  "name": "@app-speed/esbuild-meta",
  "bin": "./main.js",
  "type": "module",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "esbuild": "0.19.2",
    "yargs": "^17.7.2"
  },
  "version": "1.3.0"
}
