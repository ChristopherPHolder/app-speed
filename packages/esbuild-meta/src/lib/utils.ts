import { readFileSync, writeFileSync } from 'node:fs';
import { join, normalize } from 'node:path';
import { Metafile } from 'esbuild';

export const INVALID_FILE_PATH_ERROR_MSG = (path: string) => `No file found at ${path}`;

export function getJson<T = any>(path: string) {
  const normalizedPath = normalize(path);
  try {
    return JSON.parse(readFileSync(normalizedPath, {encoding: 'utf-8'})) as T;
  }
  catch (e) {
    throw new Error(INVALID_FILE_PATH_ERROR_MSG(normalizedPath));
  }
}

export function makeJson(path: string, file: any) {
  writeFileSync(join(...[path]), JSON.stringify(file, null, 4), {encoding: 'utf-8'});
}

export function removeDynamicImports(meta: Metafile): void {
  const chunksPaths = Object.keys(meta.outputs);
  for (const chunkPath of chunksPaths) {
    const imports = meta.outputs[chunkPath].imports;
    for (const importIndex in imports) {
      if (imports[importIndex].kind === "dynamic-import") {
        delete meta.outputs[chunkPath].imports[importIndex];
      }
    }
  }
}

const matchesPattern = (fileName: string, patterns: string[]) => patterns.some(substring => fileName.includes(substring));

export function extractEntryPoints(meta: Metafile, patterns: string[]): string[] {
  return Object.keys(meta.outputs).filter(fileName => matchesPattern(fileName, patterns));
}

export function filterMetaFromEntryPoints(meta: Metafile, entryPoints: string[]) {
  const extractedChunks = new Set<string>();
  const alreadyExtractedChildren = (chunk: string) => extractedChunks.has(chunk);
  const addToExtractedChunks = (chunk: string) => extractedChunks.add(chunk) && extractedChunks.add(`${chunk}.map`);
  const childImportedChunks = (chunk: string) => meta['outputs'][chunk]['imports'].filter(({kind}) => kind === "import-statement" || "dynamic-import").map(({path}) => path);
  function extractImport(chunks: string[]) {
    for (const chunk of chunks) {
      if (alreadyExtractedChildren(chunk)) continue;
      addToExtractedChunks(chunk);
      extractImport(childImportedChunks(chunk));
    }
  }
  function removeNotReferencedChunks() {
    const outputChunks = Object.keys(meta['outputs']);
    for (const chunk of outputChunks) {
      if (extractedChunks.has(chunk)) continue;
      delete meta['outputs'][chunk];
    }
  }

  extractImport(entryPoints);
  removeNotReferencedChunks();
}
