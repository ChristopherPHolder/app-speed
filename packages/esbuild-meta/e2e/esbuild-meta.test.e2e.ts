import { describe, it, expect, beforeAll } from 'vitest';
import { version } from '../package.json';

import { commandOutput } from './utils.js';

describe('--help', () => {
  let helpOutput: string;

  beforeAll(() => {
    helpOutput = commandOutput('npx esbuild-meta --help');
  });

  it('should show help', () => {
    expect(helpOutput).toMatchSnapshot();
  });

  it('should alias to -h', () => {
    expect(commandOutput('npx esbuild-meta -h')).toBe(helpOutput);
  });
});

describe('--version', () => {
  it('should show version', () => {
    expect(commandOutput('npx esbuild-meta --version')).toContain(version);
  });
});
