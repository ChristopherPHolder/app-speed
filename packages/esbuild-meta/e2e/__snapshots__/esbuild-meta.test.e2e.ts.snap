// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`--help > should show help 1`] = `
"esbuild-meta [command]

Commands:
  esbuild-meta filter  Filters the meta file to only include chunks required by specified entry points  [aliases: f]

Options:
      --version  Show version number  [boolean]
  -h, --help     Show help  [boolean]
"
`;
